txn.go

```diff
@@ -377,8 +377,8 @@ func (txn *Txn) modify(e *Entry) error {
 		return exceedsSize("Key", maxKeySize, e.Key)
 	case int64(len(e.Value)) > txn.db.opt.ValueLogFileSize:
 		return exceedsSize("Value", txn.db.opt.ValueLogFileSize, e.Value)
-	case txn.db.opt.InMemory && len(e.Value) > txn.db.opt.ValueThreshold:
-		return exceedsSize("Value", int64(txn.db.opt.ValueThreshold), e.Value)
+	case txn.db.opt.InMemory && int64(len(e.Value)) > txn.db.valueThreshold():
+		return exceedsSize("Value", txn.db.valueThreshold(), e.Value)
```

Comment:
Do we still need this check?

Reply:
This check is for inmemory. In this mode, the value threshold will work in a default behaviour. It adds basic validation for that mode i believe.